rust/protocol/src/storage/traits.rs

```diff
@@ -1,38 +1,59 @@
 //
-// Copyright 2020 Signal Messenger, LLC.
+// Copyright 2020-2021 Signal Messenger, LLC.
 // SPDX-License-Identifier: AGPL-3.0-only
 //
 
+//! Traits defining several stores used throughout the Signal Protocol.
+
 use async_trait::async_trait;
 use uuid::Uuid;
 
-use crate::state::{PreKeyId, SignedPreKeyId};
 use crate::{
-    IdentityKey, IdentityKeyPair, PreKeyRecord, ProtocolAddress, Result, SenderKeyRecord,
-    SessionRecord, SignedPreKeyRecord,
+    address::ProtocolAddress,
+    error::Result,
+    sender_keys::SenderKeyRecord,
+    state::{PreKeyId, PreKeyRecord, SessionRecord, SignedPreKeyId, SignedPreKeyRecord},
+    IdentityKey, IdentityKeyPair,
 };
 
+/// Handle to FFI-provided context object.
+///
+/// This object is not manipulated in rust, but is instead passed back to each FFI
+/// method invocation.
 pub type Context = Option<*mut std::ffi::c_void>;
 
+/// Each Signal message can be considered to have exactly two participants, a sender and receiver.
+///
+/// [IdentityKeyStore::is_trusted_identity] uses this to ensure the identity provided is configured
+/// for the appropriate role.
 #[derive(Debug, Clone, Eq, PartialEq)]
 pub enum Direction {
     Sending,
     Receiving,
 }
 
+/// Interface defining the TOFU identity store, which may be in-memory, on-disk, etc.
 #[async_trait(?Send)]
 pub trait IdentityKeyStore {
+    /// Return the single specific identity each store is assumed to represent, with private key.
     async fn get_identity_key_pair(&self, ctx: Context) -> Result<IdentityKeyPair>;
 
+    /// Return a [u32] specific to this store instance.
+    ///
+    /// This numerical id should be randomly generated each time the store is instantiated.
     async fn get_local_registration_id(&self, ctx: Context) -> Result<u32>;
 
+    /// Record an identity into the store. The identity is then considered "trusted".
+    ///
+    /// ah TODO: what does `Ok(false)` represent here?
```

Comment:
Good question, and definitely points to the need for docs. The return value represents whether an existing identity was replaced; if it is new or hasn't changed, the return is `Ok(false)`.

Reply:
Thanks! Fixed!